NuRadioReco/framework/event.py

```diff
@@ -587,7 +541,156 @@ def get_hybrid_information(self):
         """
         return self.__hybrid_information
 
+
+    def _register_module_event(self, instance, name, kwargs):
+        """
+        registers modules applied to this event
+
+        Parameters
+        ----------
+        instance: module instance
+            the instance of the module that should be registered
+        name: module name
+            the name of the module
+        kwargs:
+            the key word arguments of the run method
+        """
+
+        self.__modules_event.append([name, instance, kwargs])
+
+    def _register_module_station(self, station_id, instance, name, kwargs):
+        """
+        registers modules applied to this event
+
+        Parameters
+        ----------
+        station_id: int
+            the station id
+        instance: module instance
+            the instance of the module that should be registered
+        name: module name
+            the name of the module
+        kwargs:
+        Registers modules applied to a specific station within this event.
+
+        Parameters
+        ----------
+        station_id: int
+            The station id to which the module is applied.
+        instance: module instance
+            The instance of the module that should be registered.
+        name: module name
+            The name of the module.
+        kwargs:
+            The key word arguments of the run method.
+        """
+        iE = len(self.__modules_event)
+        self.__modules_station[station_id].append([iE, name, instance, kwargs])
+
+    def _iter_modules(self, station_id=None):
```

Comment:
I think making the `register_module*` methods private is a good idea, but although I'm not sure that `iter_modules` is a particularly helpful interface, I think there's some value in exposing some method that can display the modules applied to an event / station. So I'm inclined to keep this public for now.

Reply:
In my opinion its a developer interface (and actually for somewhat advanced development). Hence, I think it is okay to keep in private. If you do not know about the register_module interface you do not need to know about the iter_modules function.

Reply:
I don't agree, even if a user is never expected to explicitly write to the list of modules, retrieving it has value (e.g. in a reconstruction, if I want to find out which bandpass filters have been applied to my data). This method probably does not get used much, and there might be more intuitive ways to provide this information, but I think there should be _some_ public interface that provides this possibility.